### PR TITLE
godot-beta: Add version 4.0-beta1

### PR DESCRIPTION
**Pull request by acesyde**
Relates to #534

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/godot-beta.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "bin": "godot.exe",
+    "shortcuts": [
+        [
+            "godot.exe",
+            "Godot Engine"
+        ]
+    ],
+    "checkver": {
+        "url": "https://downloads.tuxfamily.org/godotengine/$majorVersion.$minorVersion/",
+        "regex": "beta(?<beta>[\\d.]+)"
```
Comment (tech189) [inline bucket/godot-beta.json:24]: Sadly this checkver won't work, you can't use $majorVersion or $minorVersion here. You'll have to write a checkver script to navigate the download directory (https://downloads.tuxfamily.org/godotengine/), open the latest version number folder, and then run this regex for the latest beta version. Would you like to have a go at writing it or would you rather I do it?

File: bucket/godot-beta.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "version": "4.0-beta1",
+    "description": "A feature-packed, cross-platform game engine to create 2D and 3D games from a unified interface.",
+    "homepage": "https://godotengine.org/",
+    "license": "MIT",
+    "architecture": {
+        "64bit": {
+            "url": "https://downloads.tuxfamily.org/godotengine/4.0/beta1/Godot_v4.0-beta1_win64.exe.zip"
+        },
+        "32bit": {
+            "url": "https://downloads.tuxfamily.org/godotengine/4.0/beta1/Godot_v4.0-beta1_win32.exe.zip"
```
Comment (tech189) [inline bucket/godot-beta.json:11]: Hashes are missing, you can find them here https://downloads.tuxfamily.org/godotengine/4.0/beta1/SHA512-SUMS.txt

File: bucket/godot-beta.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        "url": "https://downloads.tuxfamily.org/godotengine/$majorVersion.$minorVersion/",
+        "regex": "beta(?<beta>[\\d.]+)"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://downloads.tuxfamily.org/godotengine/$majorVersion.$minorVersion/beta$matchbeta/Godot_v$majorVersion.$minorVersion_win64.exe.zip"
+            },
+            "32bit": {
+                "url": "https://downloads.tuxfamily.org/godotengine/$majorVersion.$minorVersion/beta$matchbeta/Godot_v$majorVersion.$minorVersion_win32.exe.zip"
```
Comment (tech189) [inline bucket/godot-beta.json:32]: Add hash to autoupdate too please (after checkver is sorted)

File: bucket/godot-beta.json (hunk 1 above)
Comment (acesyde) [reply bucket/godot-beta.json:26]: Hi @tech189, thank you for your review

I have no idea how to do this script, if you can do it, I would be happy to :)

**Comment by acesyde** [comment]
@tech189 now it works like "godot-alpha.json" :)

**Pull request merged**